### PR TITLE
Add input test after install driver via installer

### PR DESCRIPTION
**Pull request by leidwang**
ID:2114623
Signed-off-by: Leidong Wang <leidwang@redhat.com>

### CODE REVIEW

**Comment by leidwang** [comment]
(1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_install.single_driver.with_vioinput.q35: STARTED
 (1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_install.single_driver.with_vioinput.q35: PASS (327.09 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2022-11-14T01.12-1907cae/results.html

**Comment by leidwang** [comment]
Hi @peixiu Could you help to review this patch?Thanks a lot!

**Comment by peixiu** [comment]
Hi @leidwang,

I hit follows error when run it on win11 guest, I tried 2 times, both failed with same error: 
python3 ConfigTest.py --guestname=Win11 --platform=x86_64 --clone=no --testcase=win_virtio_driver_install_by_installer.driver_install.single_driver.with_vioinput

(1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_install.single_driver.with_vioinput.q35: STARTED
 (1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_install.single_driver.with_vioinput.q35: ERROR: Copy file error, the detailed info:\nFile not found - virtio-win-guest-tools.exe\n0 File(s) copied\n\n. (549.07 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2022-11-21T04.31-0d2fd1f/results.html
JOB TIME   : 553.08 s

Test summary:
Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_install.single_driver.with_vioinput.q35: ERROR

**Comment by leidwang** [comment]
> Hi @leidwang,
> 
> I hit follows error when run it on win11 guest, I tried 2 times, both failed with same error: python3 ConfigTest.py --guestname=Win11 --platform=x86_64 --clone=no --testcase=win_virtio_driver_install_by_installer.driver_install.single_driver.with_vioinput
> 
> (1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_install.single_driver.with_vioinput.q35: STARTED (1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_install.single_driver.with_vioinput.q35: ERROR: Copy file error, the detailed info:\nFile not found - virtio-win-guest-tools.exe\n0 File(s) copied\n\n. (549.07 s) RESULTS : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0 JOB HTML : /root/avocado/job-results/job-2022-11-21T04.31-0d2fd1f/results.html JOB TIME : 553.08 s
> 
> Test summary: Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_install.single_driver.with_vioinput.q35: ERROR

We need use virtio-win to run the test.

**Comment by peixiu** [comment]
python3 ConfigTest.py --guestname=Win11 --platform=x86_64 --clone=no --testcase=win_virtio_driver_install_by_installer.driver_install.single_driver.with_vioinput --customsparams="cdrom_virtio = isos/windows/virtio-win-1.9.30-0.el9_1.iso"

 (1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_install.single_driver.with_vioinput.q35: STARTED
 (1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win11.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_install.single_driver.with_vioinput.q35: PASS (604.31 s)

LGTM～

**Review by peixiu — APPROVED** [review]

**Comment by leidwang** [comment]
@vivianQizhu Could you help to review this patch?Thanks a lot!

File: provider/win_driver_installer_test.py [hunk 1]
```diff
@@ -563,3 +566,84 @@ def balloon_test(test, params, vm, balloon_test_win):
...
+def vioinput_test(test, params, vm):
+    """
+    Keyboard test. Support single key and combination key tests.
+
+    :param test: kvm test object
+    :param params: dictionary with the test parameters
+    :param vm: vm object
+    """
+
+    def key_check(key):
```
Comment (vivianQizhu) [inline provider/win_driver_installer_test.py:591]: This seems exactly same with other vioinput cases, shall we abstract it into another independent module like vioinput_basic, it is okay if there is only this single one function for it; and then all the other similar steps could call the basic function.

Comment (vivianQizhu) [reply provider/win_driver_installer_test.py:591]: Move functions inside vioinput_keyboard to provider and her in win_driver_installer_test we might have one simple `from vioinput_basic import key_tap_test as vioinput_test`

Comment (leidwang) [reply provider/win_driver_installer_test.py]: Okay,sorry for the delay.Updated the code.Could you help to review it again?Thanks a lot!

**Comment by leidwang** [comment]
Hi @vivianQizhu Could you help to review it again?Thanks!

**Comment by vivianQizhu** [comment]
@leidwang Shall we remove same functions here: https://github.com/autotest/tp-qemu/blob/master/qemu/tests/vioinput_keyboard.py#L12 or it will be meaningless.

**Comment by leidwang** [comment]
Hi @vivianQizhu Could you help to review it again?Thanks!

File: provider/win_driver_installer_test.py [hunk 2]
```diff
@@ -13,9 +13,11 @@
 from virttest import utils_misc
 from virttest.utils_windows import virtio_win
 
+
```
Comment (vivianQizhu) [inline provider/win_driver_installer_test.py:16]: Seems unnecessary.

**Review by vivianQizhu — COMMENTED** [review]
LGTM otherwise. @peixiu Request for your review, thanks.

**Comment by leidwang** [comment]
Hi @peixiu Could you help to review this patch when you are free?Thanks!

**Comment by peixiu** [comment]
Great update, thank you~

LGTM~

**Review by peixiu — APPROVED** [review]

**Review by vivianQizhu — APPROVED** [review]

**Pull request merged**